lib/dfp_helper.rb

```diff
@@ -39,13 +39,13 @@ def dfp_define_slot(_i, options = {})
 </script>
       END
     end
-    
+
     def dfp_helper_head(options={single_request:true})
       return unless dfp_helper_slots.size > 0
       o = dfp_helper_slots.collect{|i|
         _targeting = (i[:targeting]||[]).collect{|k,v| ".setTargeting(#{k.to_json}, #{v.to_json})"}.join
         _slot_name = (i[:slot_name].blank?)?"":"#{i[:slot_name]} = "
-        "#{_slot_name}googletag.defineSlot('#{i[:id]}', [#{i[:size].map(&:to_s).join(', ')}], '#{i[:div_id]}').addService(googletag.pubads())#{_targeting};"
+        "#{_slot_name}googletag.defineSlot('#{i[:id]}', [#{i[:size].map(&:to_s).join(', ')}], '#{i[:div_id]}').setCollapseEmptyDiv(true).addService(googletag.pubads())#{_targeting};"
```

Comment:
so technically this will still collapse every slot; it's just being called per slot rather than globally

looking through the API documentation now, and not seeing how to get a slot object once it's already been defined ... will need to dig some more

the ramifications of collapsing these ads globally would be if there are any existing pages with ads whose style/layout expects the white space for the ad even when there is not active ad being run.